enterprise/views.py

```diff
@@ -952,3 +964,376 @@ def get(self, request, enterprise_uuid, course_id):
+    }
+
+    def get_program_prices(self, program_courses, user):
+        """
+        Get the program's price, both discounted and without discounts.
+        """
+        modes = [course['course_mode'] for course in program_courses]
+        enrolled_modes = [course['course_mode'] for course in program_courses if course['paid_for_certificate']]
+        try:
+            ecommerce_api_client = EcommerceApiClient(user)
+            program_prices = ecommerce_api_client.get_program_prices(modes, currency=None)
+            enrolled_prices = ecommerce_api_client.get_program_prices(
+                enrolled_modes,
+                currency=None
+            ) if enrolled_modes else {}
+        except HttpClientError:
+            program_prices = {}
+            enrolled_prices = {}
+            LOGGER.error("Failed to get program price details for SKUs '{skus}' for user '{username}'.".format(
+                skus=[mode['sku'] for mode in modes],
+                username=user.username,
+            ))
+
+        price = format_price(
+            program_prices.get('price', 0) - enrolled_prices.get('price', 0)
+        )
+        discounted_price = format_price(
+            program_prices.get('discounted_price', 0) - enrolled_prices.get('discounted_price', 0)
+        )
+        return (price, discounted_price) if program_prices else (None, None)
+
+    def get_program_enrollment_details(self, user, program_details):
+        """
+        Get program enrollment details, including program price (depends on enrollment status).
+
+        Encapsulate any calculation necessary to determine whether the learner is enrolled in the program and
+        checks whether the learner is enrolled in verified mode for all courses in the program. Also gets price
+        related information.
+
+        :param username: The learner who might be enrolled in the program.
+        :param program_details: The details of the program to check enrollment against.
+        :return: Whether the learner is enrolled in the program and whether they're program certificate-eligible.
+        """
+        # A single violation in the logic relevant to these will flip their booleans and keep them that way thereon.
+        enrolled_in_program = False
+        certificate_eligible = True
+
+        program_courses = []
+        enrollment_client = EnrollmentApiClient()
+        for program_course in program_details['courses']:
+            # Safely make the assumption that the first course run is available and is the one we want.
+            program_course_run = program_course['course_runs'][0]
+            course_id = program_course_run['key']
+            # Track courses so we can calculate the price later.
+            course_modes = enrollment_client.get_course_modes(course_id)
+            course_mode_for_calculation = [mode for mode in course_modes if mode['slug'] == 'verified'][0]
```

Comment:
Pretty sure I'm doing something wrong here, because I think I need to include 'professional', but not sure. Please advise!